src/cert_store.rs

```diff
@@ -243,6 +304,80 @@ impl CertStore {
             Ok(ret)
         }
     }
+
+    /// Determines if the provided cert exists in a given store
+    /// 
+    /// If the function is successful a new cert context pointing to the existing cert in the CertStore will be returned
+    pub fn find_existing_cert(&mut self, provided_cert: &CertContext) -> io::Result<CertContext> {
+        unsafe {
+            let res = Cryptography::CertFindCertificateInStore( 
+                self.0,
+                Cryptography::X509_ASN_ENCODING | Cryptography::PKCS_7_ASN_ENCODING,
+                0,                             
+                Cryptography::CERT_COMPARE_EXISTING << Cryptography::CERT_COMPARE_SHIFT,
+                provided_cert.as_inner() as *const c_void, 
+                ptr::null_mut(), 
+            ); 
+            if res.is_null() { 
+                return Err(io::Error::last_os_error());
+            } else { 
+                Ok(CertContext::from_inner(res))
+            }  
+        }   
+    }
+
+    /// Opens a .sst file and returns the contents in a CertStore
+    /// 
+    /// The file_name must a valid windows path
+    pub fn open_file(file_name: &Path) -> io::Result<CertStore> { 
+        unsafe { 
+            let data = file_name.as_os_str() 
+                .encode_wide() 
+                .chain(Some(0)) 
+                .collect::<Vec<_>>(); 
+            let store = Cryptography::CertOpenStore( 
+                Cryptography::CERT_STORE_PROV_FILENAME_W, 
+                Cryptography::CERT_QUERY_ENCODING_TYPE::default(), 
+                Cryptography::HCRYPTPROV_LEGACY::default(), 
+                0,        // dwFlags 
+                data.as_ptr() as *mut _, 
+            ); 
+            if !store.is_null() { 
+                Ok(CertStore(store)) 
+            } else { 
+                Err(io::Error::last_os_error()) 
+            } 
+        } 
+    } 
+
+    /// Exports the CertStore provided to a .sst file to the provided location
+    /// 
+    /// The file_path provided must have a .sst as an ending extension, and be a properly formatted windows path
+    pub fn create_sst(file_path: &Path, store:  &mut CertStore) -> io::Result<()> {
+        unsafe{
+            // convert store and file path to c_void pointers
+            let store_ptr = store.0 as *mut CertStore as *mut c_void;
+            let data = file_path.as_os_str() 
+                .encode_wide() 
+                .chain(Some(0)) 
+                .collect::<Vec<_>>();
+            let path_ptr = data.as_ptr() as *mut c_void;
+
+            let res = Cryptography::CertSaveStore(
+                store_ptr, 
```

Comment:
> store_ptr [](http://example.com/codeflow?start=15&length=10)

I have seen this pattern to get to the inner pointer:

store.as_inner()

might be able to remove the above store_ptr.